src/targets/gpu/include/migraphx/gpu/device/fp_conversion.hpp

```diff
@@ -0,0 +1,22 @@
+
+#ifndef MIGRAPHX_GUARD_RTGLIB_DEVICE_FP_CONVERSION_HPP
+#define MIGRAPHX_GUARD_RTGLIB_DEVICE_FP_CONVERSION_HPP
+
+#include <migraphx/argument.hpp>
+#include <migraphx/config.hpp>
+#include <hip/hip_runtime_api.h>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+argument
+fp_conversion(hipStream_t stream, const shape& output_shape, const std::vector<argument>& args);
```

Comment:
There is no need to write `output_shape` as the argument has the shape. Also, it will only take one argument as input, so there is no reason to accept a vector. This should be written as `void fp_conversion(hipStream_t stream, const argument& result, const argument& arg)`.